internal/builders/go/pkg/provenance.go

```diff
@@ -71,21 +80,35 @@ func GenerateProvenance(name, digest, command, envs, workingDir string) ([]byte,
 		return nil, err
 	}
 
-	// Generate a basic WorkflowRun for our subject based on the github
-	// context.
-	wr := slsa.NewWorkflowRun([]intoto.Subject{
-		{
-			Name: name,
-			Digest: slsa02.DigestSet{
-				"sha256": digest,
+	b := goProvenanceBuild{
+		GithubActionsBuild: slsa.NewGithubActionsBuild([]intoto.Subject{
+			{
+				Name: name,
+				Digest: slsa02.DigestSet{
+					"sha256": digest,
+				},
 			},
-		},
-	}, gh)
+		}, gh),
+	}
+
+	// Pre-submit tests don't have access to write OIDC token.
+	if isPreSubmitTests() {
+		b.GithubActionsBuild.WithClients(&slsa.NilClientProvider{})
+	}
+
+	ctx := context.Background()
+	g := slsa.NewHostedActionsGenerator(&b)
+	// Pre-submit tests don't have access to write OIDC token.
+	if isPreSubmitTests() {
+		g.WithClients(&slsa.NilClientProvider{})
+	}
+	p, err := g.Generate(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-	// Identifies that this is a slsa-framework's slsa-github-generator-go' build.
-	wr.BuildType = buildType
 	// Sets the builder specific build config.
-	wr.BuildConfig = buildConfig{
+	p.Predicate.BuildConfig = buildConfig{
```

Comment:
The idea is that BuildConfig would get set by overriding `BuildConfig` on `goProvenanceBuild`
https://github.com/slsa-framework/slsa-github-generator/blob/main/slsa/buildtype.go#L39-L40

Reply:
gotcha. I'll create a `BuildConfig()` function.

Reply:
done, PTAL